src/org/labkey/test/components/glassLibrary/grids/Pager.java

```diff
@@ -0,0 +1,194 @@
+{
+    private final WebElement _el;
+    private final WebDriver _driver;
+    //private final ResponsiveGrid _grid;
+
+    protected Pager(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public Pager jumpToPage(String jumpTo)
+    {
+        elementCache().jumpToDropdown.clickSubMenu(jumpTo);
+        return this;
+    }
+    public int getCurrentPage()
+    {
+        return Integer.parseInt(elementCache().jumpToDropdown.getButtonText());
+    }
+
+    public Pager selectPageSize(String pageSize)
+    {
+        elementCache().pageSizeDropdown.clickSubMenu(pageSize);
+        return this;
+    }
+
+    public int getPageSize()
+    {
+        return Integer.parseInt(elementCache().pageSizeDropdown.getButtonText());
+    }
+
+    public Pager clickPrevious()
+    {
+        elementCache().prevButton.click();
+        return this;
+    }
+
+    public boolean isPreviousEnabled()
+    {
+        return !elementCache().prevButton.getAttribute("disabled").equals("true");
+    }
+
+    public Pager clickNextButton()
+    {
+        elementCache().nextButton.click();
+        return this;
+    }
+
+    public boolean isNextEnabled()
+    {
+        return !elementCache().nextButton.getAttribute("disabled").equals("true");
+    }
+
+    public int start()
+    {
+        // note: the 'old' element here and in end() and total() are here to support the pager used in
+        // biologicsReportTest.testReportListPaging.  Hopefully, we can consolidate dom between that pager
+        // and the ones we now have in QueryGrid
+
+        WebElement oldStart = Locator.tagWithClass("span", "pagination-info__start")
+                .findElementOrNull(this);
+        if (oldStart != null)
+            return Integer.parseInt(oldStart.getText());
+        else
+        {
+            String value = elementCache().counts.getAttribute("data-min");
+            return Integer.parseInt(value);
+        }
+    }
+
+    public int end()
+    {
+        WebElement oldEnd = Locator.tagWithClass("span", "pagination-info__end")
+                .findElementOrNull(this);
+        if (oldEnd != null)
+            return Integer.parseInt(oldEnd.getText());
+        else
+        {
+            String value = elementCache().counts.getAttribute("data-max");
+            return Integer.parseInt(value);
+        }
+    }
+
+    public int total()
+    {
+        WebElement oldTotal = Locator.tagWithClass("span", "pagination-info__total")
+                .findElementOrNull(this);
+        if (oldTotal != null)
+            return Integer.parseInt(oldTotal.getText());
+        else
+        {
+            String value = elementCache().counts.getAttribute("data-total");
+            return Integer.parseInt(value);
+        }
+    }
```

Comment:
I confess; I'm not 100% sure what that will get us.  I agree that the current implementation is pretty verbose and would like to clean it up.  Let's talk about how this can be used

Reply:
Right now, ` summary()` (line 138) consumes the text from the 'counts' element, which only exists in the new GridPanel.